### PR TITLE
Add configuration script for alcf mira/cetus

### PR DESCRIPTION
**Pull request by dalg24**
- [x] Build DTK on alcf machine cetus/mira
- [x] Run test regression
- [x] Document

### CODE REVIEW

**Comment by aprokop** [comment]
Is related to #26

**Comment by dalg24** [comment]
@bartlettroscoe 
```
2: Test command: /usr/bin/runjob "--np" "4" "--block $BLOCKID :" "/home/lebrung/projects/dtk/mira/gcc/DataTransferKit/packages/Utils/test/DataTransferKitUtils_DBC_test.exe"
2: Test timeout computed to be: 1500
2: 2016-10-25 21:31:42.581 (INFO ) [0xfff900dc8e0] ibm.runjob.AbstractOptions: using properties file /bgsys/local/etc/bg.properties
2: unknown option block $BLOCKID :
 2/88 Test  #2: DataTransferKitUtils_DBC_test_MPI_4 ...........................................***Failed  Required regular expression not found.Regex=[End Result: TEST PASSED
]  0.06 sec
```
How do you get CTest to use an environment variable that is not yet defined at configuration time?

File: scripts/alcf_mira_cmake [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+EXTRA_ARGS=$@
+BLAS_DIR=/soft/libraries/alcf/20130312/gcc/BLAS
+LAPACK_DIR=/soft/libraries/alcf/20130312/gcc/LAPACK
+BOOST_DIR=/soft/libraries/boost/1.58.0/cnk-gcc/V1R2M2-20150523
+TRILINOS_DIR=/home/lebrung/projects/dtk/source/trilinos/12.8.1
+ARGS=(
+    -D CMAKE_BUILD_TYPE=Debug
```
Comment (aprokop) [inline scripts/alcf_mira_cmake]: Why is the build type for production runs is Debug?

Comment (dalg24) [inline scripts/alcf_mira_cmake]: I will change eventually.  I am able to build but did not manage to run the tests yet...

Comment (aprokop) [inline scripts/alcf_mira_cmake]: So this is a new untested script then?

Comment (dalg24) [inline scripts/alcf_mira_cmake]: I pushed it to be able to share the script and get help/comments.  This is a work in progress.

Comment (aprokop) [inline scripts/alcf_mira_cmake]: Sorry for being pushy. What do you need help with? As far as I can see the script itself looks fine.

Comment (dalg24) [inline scripts/alcf_mira_cmake]: I was not able to run the tests.

Comment (dalg24) [inline scripts/alcf_mira_cmake]: There is no `mpiexec` on cetus/mira.

File: scripts/alcf_mira_cmake [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    -D BUILD_SHARED_LIBS=OFF
+    -D TPL_FIND_SHARED_LIBS=OFF
+    ### SETUP TO RUN MPI TESTS ###
+    -D MPI_EXEC=runjob
+    -D MPI_EXEC_NUMPROCS_FLAG="--np"
+    -D MPI_EXEC_PRE_NUMPROCS_FLAGS=""
+    -D MPI_EXEC_POST_NUMPROCS_FLAGS="--block \$BLOCKID :"
```
Comment (dalg24) [inline scripts/alcf_mira_cmake]: This is where I tried to setup for using `runjob`.

Comment (dalg24) [inline scripts/alcf_mira_cmake]: I found the answer and replaced `\$BLOCKID` by `\$ENV{BLOCKID}`.

Comment (aprokop) [inline scripts/alcf_mira_cmake]: For future references where did you find the answer? Please point to the documentation.

Comment (dalg24) [inline scripts/alcf_mira_cmake]: No doc to point to.  I knew about [`\ENV{VAR}`](https://cmake.org/cmake/help/v3.6/variable/ENV.html) and I realized I could try to escape it just like a regular variable as I was driving home.

**Comment by bartlettroscoe** [comment]
> How do you get CTest to use an environment variable that is not yet defined at configuration time?

As far as I know, you can't.  Just reconfigure setting that var and run ctest again.

**Comment by dalg24** [comment]
> > How do you get CTest to use an environment variable that is not yet defined at configuration time?

> As far as I know, you can't. Just reconfigure setting that var and run ctest again.

Ross, I did find a solution.  See above.

**Comment by dalg24** [comment]
OK, I got all the tests to pass:
```
100% tests passed, 0 tests failed out of 88

Label Time Summary:
DataTransferKit    = 2006.23 sec

Total Test time (real) = 2006.55 sec
```
1/2 hour to run them.  A new record...

I had to run the tests using `ctest -V -j 1`.  When I tried to use more than one job, I was getting `(FATAL) [0xfff9573c8e0] CET-20400-31731-128:16829:ibm.runjob.client.Job: could not start job: block is busy`.
Running one test at a time with a latency to launch the job and to exit :(  See the log below.
```
test 87
      Start 87: DataTransferKitClassicDTKAdapters_VolumeSourceMap_test6_MPI_4

87: Test command: /usr/bin/runjob "--np" "4" "-p" "1" "--block" "CET-20400-31731-128" "--verbose=INFO" ":" "/home/lebrung/projects/dtk/mira/gcc/DataTransferKit/packages/Adapters/ClassicDTK/test/map/DataTransferKitClassicDTKAdapters_VolumeSourceMap_test6.exe"
87: Test timeout computed to be: 1500
87: 2016-11-11 17:22:13.377 (INFO ) [0xfff8abfc8e0] ibm.runjob.AbstractOptions: using properties file /bgsys/local/etc/bg.properties
87: 2016-11-11 17:22:13.379 (INFO ) [0xfff8abfc8e0] ibm.runjob.AbstractOptions: max open file descriptors: 65536
87: 2016-11-11 17:22:13.379 (INFO ) [0xfff8abfc8e0] ibm.runjob.AbstractOptions: core file limit: 18446744073709551615
87: 2016-11-11 17:22:13.381 (INFO ) [0xfff8abfc8e0] 11480:tatu.runjob.client: scheduler job id is 957289
87: 2016-11-11 17:22:13.384 (INFO ) [0xfffa43434e0] 11480:tatu.runjob.monitor: monitor started
87: 2016-11-11 17:22:13.663 (INFO ) [0xfffa43434e0] 11480:tatu.runjob.monitor: task record 1773929 created
87: 2016-11-11 17:22:13.664 (INFO ) [0xfff8abfc8e0] CET-20400-31731-128:11480:ibm.runjob.client.options.Parser: set local socket to runjob_mux from properties file
87: 2016-11-11 17:22:16.203 (INFO ) [0xfff8abfc8e0] CET-20400-31731-128:1823207:ibm.runjob.client.Job: job 1823207 started
87: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name Task 2 of 4 (0,0,0,1,0,0)  R01-M0-N04-J17 and rank 2!
87: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name Task 0 of 4 (0,0,0,0,0,0)  R01-M0-N04-J29 and rank 0!
87: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name Task 3 of 4 (0,0,0,1,1,0)  R01-M0-N04-J22 and rank 3!
87: Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name Task 1 of 4 (0,0,0,0,1,0)  R01-M0-N04-J26 and rank 1!
87:
87: ***
87: *** Unit test suite ...
87: ***
87:
87:
87: Sorting tests by group name then by the order they were added ... (time = 3.63e-06)
87:
87: Running unit tests ...
87:
87: 0. VolumeSourceMap_cylinder_test_UnitTest ... [Passed] (0.0139 sec)
87:
87: Total Time: 0.019 sec
87:
87: Summary: total = 1, run = 1, passed = 1, failed = 0
87:
87: End Result: TEST PASSED
87: 2016-11-11 17:22:20.405 (INFO ) [0xfffa43434e0] 11480:tatu.runjob.monitor: tracklib completed
87: 2016-11-11 17:22:33.401 (INFO ) [0xfff8abfc8e0] CET-20400-31731-128:1823207:ibm.runjob.client.Job: exited with status 0
87: 2016-11-11 17:22:33.401 (INFO ) [0xfff8abfc8e0] tatu.runjob.client: task exited with status 0
87: 2016-11-11 17:22:33.401 (INFO ) [0xfffa43434e0] 11480:tatu.runjob.monitor: monitor terminating
87: 2016-11-11 17:22:33.405 (INFO ) [0xfff8abfc8e0] tatu.runjob.client: monitor completed
87/88 Test #87: DataTransferKitClassicDTKAdapters_VolumeSourceMap_test6_MPI_4 .................   Passed   20.10 sec
```

**Comment by dalg24** [comment]
@aprokop @sslattery Ready for final review

File: scripts/Readme.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Building DTK on ALCF ressources
+
+## Software environment
+The MPICH wrappers of the GNU Compiler Collection (GCC) were used to build DTK on Mira/Cetus.
+Below is the corresponding SoftEnv configuration file `$HOME/.soft.mira`.
```
Comment (aprokop) [inline scripts/Readme.md]: Is .soft.mira a required name for the file? Is it something specific for ALCF or just the way you named the file?

Comment (dalg24) [inline scripts/Readme.md]: Just the way I named it.

File: scripts/Readme.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Building and testing
+```bash
+$ resoft $HOME/.soft.mira
+$ wget https://github.com/trilinos/Trilinos/archive/trilinos-release-12-8-1.tar.gz \
+  -O trilinos-release.tar.gz
+$ mkdir trilinos-release && tar -xf trilinos-release.tar.gz -C trilinos-release \
+  --strip-components=1
+$ cd trilinos-release
+$ git clone https://github.com/ORNL-CEES/DataTransferKit.git
```
Comment (aprokop) [inline scripts/Readme.md]: Maybe shallow clone (`--depth=1`)?

**Pull request merged**